core/dispatcher/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -318,10 +319,10 @@ class Invoker(
     private val LogRetryCount = 15
     private val LogRetry = 100 // millis
     private val LOG_ACTIVATION_SENTINEL = "XXX_THE_END_OF_A_WHISK_ACTIVATION_XXX"
-    private val nodejsImageName = WhiskAction.containerImageName(NodeJSExec("", None), config.dockerRegistry, config.dockerImageTag)
-    private val nodejs6ImageName = WhiskAction.containerImageName(NodeJS6Exec("", None), config.dockerRegistry, config.dockerImageTag)
-    private val swiftImageName = WhiskAction.containerImageName(SwiftExec(""), config.dockerRegistry, config.dockerImageTag)
-    private val swift3ImageName = WhiskAction.containerImageName(Swift3Exec(""), config.dockerRegistry, config.dockerImageTag)
+    private val sentinelledActions = {
```

Comment:
The creation of this set looks like a big hack to me. Can we replace by a `def` that checks the type and use that instead of `.contains` below?